### PR TITLE
Make Endpoints deployable

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

When a `Service` depends on an `Endpoint` (for non-trashable environments), they must share the same name to be usable by `Pod`s. The counterpart is that some DeploymentConfigs related to this service should only be created in trashable environments. The current work improves the situation.

## Proposal

- [x] ignore empty rendered templates
- [x] make endpoints deployable (B/G)
- [x] update `richie` templates following the aforementioned conventions
- [x] update `edxapp` templates following the aforementioned conventions

### CODE REVIEW

File: group_vars/customer/patient0/development/configs/edxapp/cms/docker_run.py.j2 [hunk 1]
```diff
@@ -1 +1,74 @@
...
+        "USER": config("MYSQL_USER"),
+        "PASSWORD": config("MYSQL_PASSWORD"),
+    }
+}
+
+CONTENTSTORE = {
+    "DOC_STORE_CONFIG": {
+        "collection": "modulestore",
+        "db": config("MONGODB_DATABASE"),
+        "host": "edxapp-mongodb-{{ deployment_stamp }}.{{ project_name }}.svc.cluster.local",
```
Comment (sampaccoud) [inline group_vars/customer/patient0/development/configs/edxapp/cms/docker_run.py.j2]: modify(CONTENTSTORE, lambda o: o["DOC_STORE_CONFIG"]["host"] = "edxapp-mongodb-{{ deployment_stamp }}.{{ project_name }}.svc.cluster.local"

File: group_vars/customer/patient0/development/configs/edxapp/cms/settings.yml.j2 [hunk 2]
```diff
@@ -2,14 +2,16 @@
 
 PLATFORM_NAME: "{{ project_display_name }}"
 
+SITE_NAME: "{{ project_display_name }}"
```
Comment (sampaccoud) [inline group_vars/customer/patient0/development/configs/edxapp/cms/settings.yml.j2]: The SITE_NAME is used as the domain to build absolute urls included in emails...

You should remove all mentions to it in Arnold as it is now configured with a good default in fun-platform: 
https://github.com/openfun/fun-platform/blob/master/config/lms/docker_run_production.py#L193

File: group_vars/customer/patient0/development/configs/edxapp/lms/settings.yml.j2 [hunk 3]
```diff
@@ -2,14 +2,16 @@
 
 PLATFORM_NAME: "{{ project_display_name }}"
 
+SITE_NAME: "{{ project_display_name }}"
```
Comment (sampaccoud) [inline group_vars/customer/patient0/development/configs/edxapp/lms/settings.yml.j2]: ditto

File: tasks/get_objects_for_app.yml [hunk 4]
```diff
@@ -2,13 +2,31 @@
 
 - name: Set templates list for this app
   set_fact:
-    templates: "{{ app | json_query('services[*].templates[]') | list }}"
+    raw_templates: "{{ app | json_query('services[*].templates[]') | list }}"
   tags: deploy
 
+# Remove empty rendered templates (e.g. inactivated for an env_type)
+- name: Ignore empty rendered templates
```
Comment (sampaccoud) [inline tasks/get_objects_for_app.yml:9]: :crying_cat_face:

Comment (jmaupetit) [reply tasks/get_objects_for_app.yml:9]: Yes, we should work on this...

Comment (jmaupetit) [reply tasks/get_objects_for_app.yml:9]: I've declared an issue for this pain point, see #128

**Review by sampaccoud — APPROVED** [review]

File: core_apps/acme/templates/app/role.yml.j2 [hunk 5]
```diff
@@ -50,16 +50,3 @@ rules:
   - update
```
Comment (sampaccoud) [inline core_apps/acme/templates/app/role.yml.j2:50]: in commit message: s/loose/lose

**Pull request merged**